docs/learn/vadalog/02-language-primitives.md

```diff
@@ -82,59 +68,82 @@ following:
 % this is a comment
 ```
 
+## Data types
+
+The table below shows the data types supported by Vadalog, along with the
+literals for the respective constants.
+
+| Data type | Examples of constant literals             |
+| --------- | ----------------------------------------- |
+| string    | `"string literal"`, `"a string"`, `""`    |
+| integer   | `1`, `3`, `5`, `-2`, `0`                  |
+| double    | `1.22`, `1.0`, `-2.3`, `0.0`              |
+| date      | `2012-10-20`, `2013-09-19 11:10:00`       |
+| boolean   | `#T`, `#F`                                |
+| set       | `{1}`, `{1,2}`, `{}`, `{"a"}`, `{2.0,30}` |
+| list      | `[1]`, `[1,2]`, `[]`, `["a"]`, `[2.0,30]` |
+| unknown   | `/`                                       |
+
 ## Variables
 
 Variables have different interpretations in different programming paradigms. In
 imperative languages, they are essentially memory locations that may have some
 contents and the contents may change over time.
 
 In algebra or physics, a variable represents a concrete value and its function
-is somewhat similar to a pronoun in natural language. In effects, once we
-replace variables with concrete actual values, we have relations between
+is somewhat similar to a pronoun in natural language. If we were to
+replace such variables with actual concrete values, we will have relations between
 concrete arithmetic expressions.
 
 Variables in Vadalog are more like variables in algebra than like those of
-imperative programming languages.
-
-Specifically, a good interpretation is the following.
-
-**Variables in Vadalog are like variables in first-order logic**.
+imperative programming languages--they are more like variables in first-order logic.
 
 For example, consider the following statements:
 
-- "For any man X there exists a father Y"
 - "Every father X is a man"
+- "For any man X there exists a father Y"
 
 These statements can be true or false depending on how we choose to instantiate
-X and Y, which means, what specific concrete values we choose. There are
-quantifiers "for any", "every" (or "for all"), namely universal quantification,
-and "there exists", namely existential quantification.
+X and Y, which means, what specific concrete values we choose.
+
+The variables in the body are "universally quantified" (∀), which means that the predicate is true **for all** possible values of such variables.
+
+If a variable appear in the head, but not in the body, it is
+"existentially quantified" (∃), which means **there exists** at least one value of the variable that makes the head true.
+
+In the following example, `X` is existentially quantified. There exists one `X`, namely "Luke Skywalker", who has a father, Y.
```

Comment:
It's not 100% clear to me if we should include this -- this might actually be more confusing, since it's not really something that the writer can affect, and is more of a description of the nature of variables.

Perhaps instead we should include the idea that you can add variables to the head of a rule even though it doesnt exist in the body.

Reply:
As per the yesterday's discussion here are some definitions.

An atom has the form p(t1, . . . , tn), where p is an n-ary predicate, and ti’s are terms. For an atom a, dom(a) and var (a) is the set of its terms and variables, respectively;

So,

1.  An Atom is like an instance of a predicate with a certain ordered list of Terms (variables and constants). e.g: a(X,Y,Z)
2. A Fact is like an instance of an Atom with a certain ordered list of Constants. e.g a(Tesla,Amazon,0.5).

Reply:
Regarding your point on existential quantification, we agree that we could simply go for being able to add variables to the head to represent an unknown concept